InscryptionAPI/Triggers/CustomTriggerPatches.cs

```diff
@@ -172,43 +160,253 @@ private static IEnumerator TriggerDeathTriggers(IEnumerator result, PlayableCard
+            bool itemCanBeUsed = true;
+            string consumableName = __instance?.Consumable?.Data?.name;
+            if (!string.IsNullOrEmpty(consumableName))
+            {
+                CustomTriggerFinder.CollectDataAll<IItemCanBeUsed, bool>(false, x => x.RespondsToItemCanBeUsed(consumableName, itemCanBeUsed), x => itemCanBeUsed = x.CollectItemCanBeUsed(consumableName, itemCanBeUsed));
+            }
+            if (itemCanBeUsed)
+            {
+                bool successInActivation = false;
+                Type activationtype = __instance.Consumable.ActivateSequence().GetType();
+                if (!string.IsNullOrEmpty(consumableName))
+                {
+                    yield return CustomTriggerFinder.TriggerAll<IOnPreItemUsed>(false, x => x.RespondsToPreItemUsed(consumableName, __instance is HammerItemSlot), x =>
+                        x.OnPreItemUsed(consumableName, __instance is HammerItemSlot));
+                }
+                while (result.MoveNext())
+                {
+                    yield return result.Current;
+                    if (result.Current.GetType() == activationtype)
+                    {
+                        if (!string.IsNullOrEmpty(consumableName) && __instance.Consumable != null)
                         {
-                            if (i.RespondsToOtherCardDieInHand(__instance, slotBeforeDeath, wasSacrifice, killer))
-                                yield return i.OnOtherCardDieInHand(__instance, slotBeforeDeath, wasSacrifice, killer);
+                            successInActivation = !__instance.Consumable.ActivationCancelled;
                         }
                     }
                 }
+                if (!string.IsNullOrEmpty(consumableName))
+                {
+                    yield return CustomTriggerFinder.TriggerAll<IOnPostItemUsed>(false, x => x.RespondsToPostItemUsed(consumableName, successInActivation, __instance is HammerItemSlot), x =>
+                        x.OnPostItemUsed(consumableName, successInActivation, __instance is HammerItemSlot));
+                }
+            }
+            else
+            {
+                __instance?.Consumable?.PlayShakeAnimation();
+                if (!string.IsNullOrEmpty(consumableName))
+                {
+                    yield return CustomTriggerFinder.TriggerAll<IOnItemPreventedFromUse>(false, x => x.RespondsToItemPreventedFromUse(consumableName), x => x.OnItemPreventedFromUse(consumableName));
+                }
+            }
+            yield break;
+        }
+
+        [HarmonyPatch(typeof(PlayableCard), nameof(PlayableCard.GetPassiveAttackBuffs))]
+        [HarmonyPostfix]
+        private static void PassiveAttackBuffs(PlayableCard __instance, ref int __result)
+        {
+            int dummyResult = __result;
+            CustomTriggerFinder.CollectDataAll<IOnCardPassiveAttackBuffs, int>(true, x => x.RespondsToCardPassiveAttackBuffs(__instance, dummyResult), x => dummyResult =
```

Comment:
Looking through this, I believe this will need to check if the playable card instance is on the board, correct?